pkg/ottl/contexts/ottlresource/resource.go

```diff
@@ -38,6 +38,11 @@ func (tCtx TransformContext) getCache() pcommon.Map {
 	return tCtx.cache
 }
 
+// [TODO]: Find appropriate way to access schemaURLItem using resource
```

Comment:
My suggestion for solving this would be to move `SchemaURLItem` somewhere where it is exported, then add a new `SchemaURLItem` parameter to `NewTransformContext`. Callers can then pass in the pdata object for whatever signal they are working with.